Note that this is not working 100% yet:
![CleanShot 2024-09-23 at 11 15 50](https://github.com/user-attachments/assets/a4a3789c-634c-432a-b1d5-a0720faa51c2)

```
Run python manage.py migrate
  python manage.py migrate
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.11.10/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.11.10/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.10/x64
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.10/x64
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.10/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.11.10/x64/lib
python: can't open file '/home/runner/work/django-auto-actions/django-auto-actions/manage.py': [Errno 2] No such file or directory
Error: Process completed with exit code 2.
```

Also, going forward with this implementation, you will need to create an account and connect [Codecov](https://about.codecov.io/) to this repository (free for open source projects). There are other options out there if you prefer, I just went ahead with this one ;)

edit: I removed the `migrate` part and workflow reaches the tests now :+1:

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -31,6 +31,12 @@ jobs:
         with:
           python-version: '3.11'
 
+      # Install PostgreSQL client
+      - name: Install PostgreSQL client tools
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y postgresql-client
+
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
@@ -42,10 +48,6 @@ jobs:
             echo "Waiting for postgres..."
             sleep 1
           done
-
-      - name: Run migrations
-        run: |
-          python manage.py migrate
 
       - name: Run tests with coverage
         run: |

--- a/README.md
+++ b/README.md
@@ -68,6 +68,19 @@ docker compose up -d
 python runtests.py
 ```
 
+## Running ruff (linting & formatting)
+
+```bash
+pip install -r requirements.txt
+ruff check
+```
+
+You can also run `ruff fix` to automatically fix some of the issues.
+
+```bash
+ruff check --fix
+```
+
 ## Support & Contributing
 
 If you like it, please consider giving this project a star. If you’re using the package, let me know! You can also [create an issue](https://github.com/Flexonze/django-auto-actions/issues/new) for any problems or suggestions. PRs are always welcome!
@@ -75,3 +88,7 @@ If you like it, please consider giving this project a star. If you’re using th
 ## Authors
 
 - **Félix Gravel** — [@Flexonze](https://www.github.com/flexonze)
+
+## License
+
+[MIT](LICENSE.txt) © [Félix Gravel](https://github.com/Flexonze)

--- a/runtests.py
+++ b/runtests.py
@@ -74,12 +74,15 @@ def runtests():
     cov.save()
     cov.report()
 
-    # Delete the generated migrations folder
+    # Safely delete the generated migrations folder
     BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
     folder = os.path.join(
         BASE_DIR, "django_auto_actions", "django_auto_actions", "migrations"
     )
-    shutil.rmtree(folder)
+    try:
+        shutil.rmtree(folder)
+    except FileNotFoundError:
+        print(f"Folder {folder} does not exist. Skipping deletion.")
 
     sys.exit(bool(failures))
 

